pkg/server/grpc/option.go

```diff
@@ -0,0 +1,42 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"time"
+)
+
+type options struct {
+	gracePeriod time.Duration
+	listen      string
+
+	tlsConfig *tls.Config
+}
+
+// Option overrides behavior of Server.
+type Option interface {
+	apply(*options)
+}
+
+type optionFunc func(*options)
+
+func (f optionFunc) apply(o *options) {
+	f(o)
+}
+
+func WithGracePeriod(t time.Duration) Option {
```

Comment:
Nit: these are clear to us, but just to be strict, it would be nice to comment them